src/algorithms/synthesis/encoding.cpp

```diff
@@ -0,0 +1,126 @@
+#include "algorithms/synthesis/encoding.hpp"
+
+namespace syrec {
+
+    auto extend(TruthTable& tt) -> void {
+        // ensure that the resulting complete table can be stored in the cube map (at most 63 inputs, probably less in practice)
+        assert(tt.cubeMap.empty() || (tt.nInputs() <= std::log2(tt.cubeMap.max_size()) && tt.nInputs() <= 63U));
+
+        if (!tt.cubeMap.empty() && (tt.nInputs() > static_cast<std::size_t>(std::log2(tt.cubeMap.max_size())) || tt.nInputs() > 63U))
+            throw std::invalid_argument("Overflow!, Number of inputs more than 64");
```

Comment:
This error message is wrong. 
First of all, one of the cases where this triggers is when there are more than _63_ inputs. And there is also a second case (the first condition in the if statement).
Print a proper error message here and please also cover that in a test case.

Reply:
Apart from changing the error message to 63 inputs and removing the empty condition. what else should I do? I'm a bit confused here.

Reply:
Nothing needs to be removed at all.
There are simply to conditions that can trigger the error, i.e. `tt.nInputs() > static_cast<std::size_t>(std::log2(tt.cubeMap.max_size())` and `tt.nInputs() > 63U`.
The error message should reflect the proper value that was the cause of the exception.
Specifically whichever of `static_cast<std::size_t>(std::log2(tt.cubeMap.max_size())` and `63` is smaller.

Reply:
done 👍🏼